src/components/common/labelFor.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable */
+import rdf from 'rdf-ext'
+import * as ns from '../../namespace.js'
+import { cubeDimensionsWithFallBack, toViewDimension } from './cubeDimensions.js'
+
+function isLabelDimension (dimension) {
+  return !!dimension.ptr.has(ns.view.labelFor).term
+}
+
+function getLabelCubeDimension ({
+  dimension,
+  view
+}) {
+  const labelDimension = dimension.ptr.in(ns.view.labelFor).term
+  if (labelDimension) {
+    const viewLabelDimension = view.dimensions.find(dimension => {
+      return dimension.ptr.term.equals(labelDimension)
+    })
+    return viewLabelDimension ? cubeDimensionsWithFallBack({ dimension: viewLabelDimension })[0] : undefined
+  }
+}
+
+function getEffectiveDimensions ({ view }) {
```

Comment:
Returns the columns defined in the projection, and filters out the ones specified in view:labelFor